### PR TITLE
explicitApi + saveable ViewModels

### PR DESCRIPTION
**Pull request by vkatz**
- [x]  #52 (https://kotlinlang.org/docs/whatsnew14.html#explicit-api-mode-for-library-authors) 
- [x]  #49 

Fix multiple classes visibility issue (was unintended visible)

### CODE REVIEW

File: example/app/composeApp/build.gradle.kts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import org.jetbrains.compose.desktop.application.dsl.TargetFormat
 import org.jetbrains.kotlin.gradle.ExperimentalKotlinGradlePluginApi
+import org.jetbrains.kotlin.gradle.ExperimentalWasmDsl
 import org.jetbrains.kotlin.gradle.dsl.JvmTarget
-import org.jetbrains.kotlin.gradle.targets.js.dsl.ExperimentalWasmDsl
 import org.jetbrains.kotlin.gradle.targets.js.webpack.KotlinWebpackConfig
 
+
```
Comment (egorikftp) [inline example/app/composeApp/build.gradle.kts:7]: empty can be removed

File: example/app/composeApp/src/commonMain/kotlin/composegears/tiamat/example/ViewModels.kt [hunk 2]
```diff
@@ -76,25 +84,23 @@ val ViewModelsScreen1 by navDestination<Unit> {
...
+    val sharedViewModel = rememberSharedViewModel(provider = ::SharedViewModel)
 
     SimpleScreen("ViewModel's - Screen 2") {
         Column(
             modifier = Modifier.padding(16.dp),
             horizontalAlignment = Alignment.CenterHorizontally
         ) {
             val sharedTimerValue by sharedViewModel.timer.collectAsState()
-            ViewModelInfo(hashCode = sharedViewModel.hashCode(), timer = sharedTimerValue)
+            ViewModelInfo("SharedViewModel", hashCode = sharedViewModel.hashCode(), timer = sharedTimerValue)
```
Comment (egorikftp) [inline example/app/composeApp/src/commonMain/kotlin/composegears/tiamat/example/ViewModels.kt:95]: can be

```kotlin
ViewModelInfo(
      name = "SharedViewModel", 
      hashCode = sharedViewModel.hashCode(), 
      timer = sharedTimerValue
)
```

File: example/sample-koin/src/commonMain/kotlin/composegears/tiamat/sample/koin/viewmodel/SaveableViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+
+internal class SaveableViewModel(savedState: SavedState?) : TiamatViewModel(), Saveable {
+    private val _timer = MutableStateFlow(0)
+    val timer = _timer.asStateFlow()
+
+    init {
+        _timer.value = savedState?.get("count") as? Int? ?: 0
```
Comment (egorikftp) [inline example/sample-koin/src/commonMain/kotlin/composegears/tiamat/sample/koin/viewmodel/SaveableViewModel.kt:17]: `savedState?.getOrDefault("count",  0)`?

File: example/sample-koin/src/commonMain/kotlin/composegears/tiamat/sample/koin/KoinIntegrationScreen.kt [hunk 4]
```diff
@@ -33,15 +35,19 @@ val KoinIntegrationScreen by navDestination<Unit> {
...
     SimpleScreen("Koin integration") {
         Column(
             horizontalAlignment = Alignment.CenterHorizontally,
             verticalArrangement = Arrangement.spacedBy(8.dp)
         ) {
-            val timer by sharedViewModel.timer.collectAsState()
-            ViewModelInfo(hashCode = sharedViewModel.hashCode(), timer = timer)
+            val timer1 by sharedViewModel.timer.collectAsState()
+            val timer2 by saveableViewModel.timer.collectAsState()
+            ViewModelInfo("SharedViewModel", hashCode = sharedViewModel.hashCode(), timer = timer1)
```
Comment (egorikftp) [inline example/sample-koin/src/commonMain/kotlin/composegears/tiamat/sample/koin/KoinIntegrationScreen.kt:48]: the same and below, can be with named args to make code consistent

Comment (vkatz) [reply example/sample-koin/src/commonMain/kotlin/composegears/tiamat/sample/koin/KoinIntegrationScreen.kt:48]: Below one is not "shared" it is "saveable" - another name

File: example/app/composeApp/build.gradle.kts (hunk 1 above)
Comment (vkatz) [reply example/app/composeApp/build.gradle.kts:7]: done

File: example/sample-koin/src/commonMain/kotlin/composegears/tiamat/sample/koin/viewmodel/SaveableViewModel.kt (hunk 3 above)
Comment (vkatz) [reply example/sample-koin/src/commonMain/kotlin/composegears/tiamat/sample/koin/viewmodel/SaveableViewModel.kt:17]: image
Not supported for some targets

File: example/app/composeApp/src/commonMain/kotlin/composegears/tiamat/example/ViewModels.kt (hunk 2 above)
Comment (vkatz) [reply example/app/composeApp/src/commonMain/kotlin/composegears/tiamat/example/ViewModels.kt:95]: https://github.com/ComposeGears/Tiamat/pull/68#discussion_r1773678643

**Review by egorikftp — APPROVED** [review]

**Pull request merged**